### PR TITLE
fix: Currency input text overlap

### LINKED ISSUE #7083: [Bug] Currency Input Overlap Text
## Description
Using CNY currency type overlaps with the text in an input
https://downloads.intercomcdn.com/i/o/384523291/a065f95e40447f9bf224463b/image.png?expires=1630655323&signature=796e0d43405d475daac3c1709bc3f002d5eac0bb2451cf57d692dd7ce4f67321

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Layout for InputWidget is changed to make it flexible.

Fixes #7083 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Tested manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/currency-input-layout 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.8 **(0)** | 36.55 **(-0.01)** | 33.93 **(0)** | 55.35 **(0)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 38.26 **(-0.87)** | 36.21 **(0)** | 55.08 **(-0.27)**
 :green_circle: | app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx | 51.85 **(1.85)** | 11.11 **(1.11)** | 25 **(2.78)** | 50 **(1.85)**
 :red_circle: | app/client/src/widgets/InputWidget/component/index.tsx | 58.46 **(-0.6)** | 45.35 **(-0.43)** | 60.47 **(-2.94)** | 57.14 **(-0.58)**</details>

### CODE REVIEW

**Review by somangshu — DISMISSED** [review]
Good implementation.

One UI feedback, the number in the input do not looked aligned to the country code
Screenshot 2021-09-16 at 4 54 30 PM

File: app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx [hunk 1]
```diff
@@ -14,8 +14,8 @@ const DropdownTriggerIconWrapper = styled.div<{ disabled?: boolean }>`
   align-items: center;
   justify-content: space-between;
   font-size: 14px;
-  height: ${(props) => (props.disabled ? 32 : 19)}px;
-  line-height: ${(props) => (props.disabled ? 32 : 19)}px;
+  height: ${(props) => (props.disabled ? 32 : 18)}px;
```
Comment (somangshu) [inline app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:17]: Why different height based on if its disabled?

**Review by somangshu — DISMISSED** [review]
@aswathkk one small comment for change

Approving the run the tests

File: app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx (hunk 1 above)
Comment (aswathkk) [reply app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:17]: I also didn't get it. It was there before, so I just modified the second part.
32px height is found to be working without any issue in disabled or normal mode.

Comment (somangshu) [reply app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:17]: @vicky-primathon can you help us out here, Whats the need to this?

File: app/client/src/widgets/InputWidget/component/index.tsx [hunk 2]
```diff
@@ -177,7 +177,8 @@ const StyledNumericInput = styled(NumericInput)`
     > {
       &:first-child:not(input) {
         position: static;
-        background: #fff;
+        background: ${(props) =>
+          props.disabled ? "rgba(206, 217, 224, 0.5)" : "#fff"};
```
[the remark this answers is not in the document]
Comment (aswathkk) [reply app/client/src/widgets/InputWidget/component/index.tsx:181]: This is the color that is assigned from Blueprintjs.
I couldn't find it anywhere else in the codebase.

File: app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx (hunk 1 above)
Comment (vicky-primathon) [reply app/client/src/widgets/InputWidget/component/ISDCodeDropdown.tsx:17]: @aswathkk  @somangshu  we can remove condition for disabled here, 32px height works in both cases.

File: app/client/src/widgets/InputWidget/component/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (aswathkk) [reply app/client/src/widgets/InputWidget/component/index.tsx:181]: @somangshu Can I make use of `Colors.SELECT_DISABLED` which is pointing to the same color in hex?
Or should I create another constant for the disabled input?

[the remark this answers is not in the document]
Comment (somangshu) [reply app/client/src/widgets/InputWidget/component/index.tsx:181]: @aswathkk that sounds acceptable, We just need to make sure that the disabled state is consistent with the input

**Review by somangshu — APPROVED** [review]
LGTM

**Review by somangshu — APPROVED** [review]
LGTM

**Pull request merged**